### PR TITLE
Complete the stateful API & UI integration & Storage Account support

### PR DESCRIPTION
**Pull request by james-tn**
1. Add session state management to Agent Factory
2. Restore history from session_id
3. Integration streamlit front-end with stateful API
4. Moving the content integration to storage account

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -5,7 +5,10 @@
 			"version": "3.11"
 		}
 	},
-	"initializeCommand": "pip install poetry && poetry lock --no-update && poetry install",
+<<<<<<< HEAD
```
Comment (Tyler-R-Kendrick) [inline .devcontainer/devcontainer.json]: Merge conflict code will cause the project to break. Make sure to resolve this before checking in.

File: src/app/app.py [hunk 2]
```diff
@@ -50,16 +56,16 @@
     st.title(body='Deep RAG AI Copilot')
     st.markdown(body='''
     ''')
-    st.checkbox(label="Show AI Assistant's internal thought process",
-                key='show_internal_thoughts', value=False)
+    # st.checkbox(label="Show AI Assistant's internal thought process",
```
Comment (Tyler-R-Kendrick) [inline src/app/app.py]: Please remove commented out code.

File: src/functions/search_vector_function.py [hunk 3]
```diff
@@ -1,71 +1,89 @@
...
+            storage_account_key: str,  
+            storage_account_name: str,  
+            container_name: str  
+        ) -> None:  
+        self.__logger: Logger = logger  
+        self.__search_client: SearchClient = search_client  
+        self.__client: AzureOpenAI = client  
+        self.__model: str = model  
+        self.__image_directory: str = image_directory  
+        self.__blob_service_client = BlobServiceClient(
```
Comment (Tyler-R-Kendrick) [inline src/functions/search_vector_function.py:33]: This is built-in behavior that already works. No need to recreate it.

**Review by Tyler-R-Kendrick — CHANGES_REQUESTED** [review]
There are some changes that prevent execution here. If those are changed, I'll approve the merge into main.

File: src/functions/search_vector_function.py (hunk 3 above)
Comment (james-tn) [reply src/functions/search_vector_function.py:33]: @Tyler-R-Kendrick can you then provide the authentication configuration in fss? I don't see the logic for it so I had to implement blob client method

Comment (Tyler-R-Kendrick) [reply src/functions/search_vector_function.py:33]: Its done through environment variables. You can see examples of which to set here:
https://github.com/fsspec/adlfs?tab=readme-ov-file#setting-credentials
I'll fix it in a followup checkin.

**Review by Tyler-R-Kendrick — APPROVED** [review]

**Pull request merged**